### PR TITLE
Add some codes for explainer

### PR DESCRIPTION
**Pull request by mapleFU**
Signed-off-by: mapleFU <1506118561@qq.com>

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/elle/core/core.go [hunk 1]
```diff
@@ -11,6 +11,18 @@ const (
 	Realtime Rel = "realtime"
 )
 
+type DependType int
+
+const (
+	RealtimeDepend DependType = iota
```
Comment (mahjonp) [inline pkg/elle/core/core.go:17]: Do not use iota and int type, it's hard to read.

File: pkg/elle/core/core.go [hunk 2]
```diff
@@ -24,7 +36,7 @@ func MergeAnomalies(anomalies ...Anomalies) Anomalies {
 type DataExplainer interface {
 	// Given a pair of operations a and b, explains why b depends on a, in the
 	//    form of a data structure. Returns `nil` if b does not depend on a.
-	ExplainPairData() interface{}
+	ExplainPairData(p1, p2 PathType) ExplainResult
```
Comment (mahjonp) [inline pkg/elle/core/core.go:39]: PathType seems always be core.Op type? IMO we can use core.Op here directly.

File: pkg/elle/core/core.go [hunk 3]
```diff
@@ -34,23 +46,41 @@ func CombineExplainer([]DataExplainer) DataExplainer {
...
+	Path []PathType
 }
 
 // TODO: refine me
-type Step struct{}
+type Step struct {
+	Result ExplainResult
+}
+
+func explainCyclePairData(pairExplainer DataExplainer, prePath PathType, currentPath PathType) ExplainResult {
```
Comment (mahjonp) [inline pkg/elle/core/core.go:61]: ditto. use core.Ops may better.

File: pkg/elle/core/core.go (hunk 1 above)
Comment (mapleFU) [reply pkg/elle/core/core.go:17]: Emm what is better here?

Comment (mahjonp) [reply pkg/elle/core/core.go:17]: Perhaps the string literal is better

File: pkg/elle/core/core.go (hunk 2 above)
Comment (mapleFU) [reply pkg/elle/core/core.go:39]: OK

**Pull request merged**